chainer/dataset/convert.py

```diff
@@ -139,3 +145,106 @@ def _concat_arrays_with_padding(arrays, padding):
             result[(i,) + slices] = src
 
     return result
+
+
+class ConcatWithAsyncTransfer(object):
+
+    """Interface to concatenate data and transfer them to GPU asynchronously.
+
+    It enables to transfer next batch of input data to GPU while GPU is
+    running kernels for training using current batch of input data.
+    """
+
+    def __init__(self, stream=None):
+        self.stream = stream
+        self.pin_arrays = [[], []]  # numpy arrays with pinned memory
+        self.cp_arrays = [[], []]  # cupy arrays
```

Comment:
If these two attributes are not a part of "public" API, add `_` to the prefix of the names.

Reply:
OK, will change in that way.